smdebug/exceptions.py

```diff
@@ -10,15 +10,23 @@ class SMDebugCustomerError(SMDebugError):
     pass
 
 
-class SMDebugTypeError(SMDebugError):
+class SMDebugTypeError(SMDebugError, TypeError):
     pass
 
 
-class SMDebugRuntimeError(SMDebugError):
+class SMDebugRuntimeError(SMDebugError, RuntimeError):
     pass
 
 
-class InvalidCollectionConfiguration(Exception):
+class SMDebugNotImplementedError(SMDebugError, NotImplementedError):
+    pass
+
+
+class SMDebugValueError(SMDebugError, ValueError):
+    pass
+
+
+class InvalidCollectionConfiguration(SMDebugError):
```

Comment:
Are we going to add `SMDebug` before `InvalidCollectionConfiguration` here?

Reply:
seems unnecessary since the telemetry is going to catch the exceptions from the class SMDebugError, not from the name of the exception